<launch>
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="num_of_waypoints" default="5"/>
    
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
        <arg name="model" value="$(arg model)"/>
    </include>
    
    <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_empty_world.launch">
        <arg name="model" value="$(arg model)"/>
    </include>
    
    <!--node pkg="path_following" name="robot_params_server" type="reconfig_server.py" args="turtlebot3_$(arg model)" output="screen"/-->
    <node pkg="rqt_gui" name="rqt_gui" type="rqt_gui" args="-s reconfigure"/>
    
    <node pkg="path_following" name="path_follow" type="path_follow.py" output="screen" args="turtlebot3_$(arg model)" required="true">
        <param name="Kp_lin" value="1.0"/>
        <param name="Ki_lin" value="0.0"/>
        <param name="Kd_lin" value="0.0"/>
        <param name="Kp_ang" value="3.0"/>
        <param name="Ki_ang" value="0.4"/>
        <param name="Kd_ang" value="0.02"/>
        <param name="I_lin_windup_limit" value="10.0"/>
        <param name="I_ang_windup_limit" value="10.0"/>
        <param name="max_lin_vel" value="0.5"/>
        <param name="max_ang_vel" value="1.0"/>
        <param name="setpoint_tolerance" value="0.1"/>
    </node>
    
    <node pkg="path_following" name="set_waypoints" type="set_waypoints.py" output="screen" args="$(arg num_of_waypoints)"/>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find path_following)/rviz/path_follow.rviz" required="true"/>
</launch>

